### ISSUE TITLE
How to distribute Android application via Play Store?

### ISSUE THREAD

**Issue by benadrasni** [flutter/flutter#30213]
I have a problem to create and distrubute single or multiple apks to cover maximum devices, What I tried so far:

1. "flutter build appbundle"
- until March 18th when I upload bundle to the Play Store, it creates apks only for armeabi-v7a and I didn't get any error from Crashlytics about missing .so files. Application was working on my Pixel 2 XL and Nexus 6P.
- from March 19th when I upload bundle to the Play store it creates apks for all platforms ('armeabi', 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'), but only apks for armeabi-v7a includes libflutter.so. Application start crashing on devices which installs apks for different platforms. I've got many errors from Crashlytics about missing libflutter.so as well as libc.so, libart.so. Application was working on my Pixel 2 XL, but crashing on Nexus 6P.

2. "flutter build apk"
- I have tried it only after bundle failed. I've got very same results as bundle after March 19th

3. 2 apks: "flutter build apk --release --target-platform=android-arm64" with platform restriction to arm64-v8a and "flutter build apk" for other platforms
-  Application start crashing on devices with platform 'x86' and 'x86_64'. I've got errors from Crashlytics about missing libflutter.so as well as libc.so, libart.so. Maybe it was crashing for other platforms as well but I didn't check it, for x86 it was obvious.
- At one point I've created 4 apks, one for each platform except 'armeabi'. I have used libflutter.so from /bin/cache/artifacts/engine/android-x86/ resp. /bin/cache/artifacts/engine/android-x64/. It doesn't help.

which brings me to latest solution:

4. 2 apks: "flutter build apk --release --target-platform=android-arm64" with platform restriction to arm64-v8a and "flutter build apk" with platform restriction to 'armeabi-v7a'. It seems to be working for most devices, but so far I've got several reports about missing .so files, some devices have installed 'arm64-v8a' some 'armeabi-v7a':
- NOKIA 6 TA-1021, Android 9  - missing libc.so, libflutter.so
- Xperia XZ1 SOV36, Android 9 - missing libc.so, libflutter.so
- Redmi Note 5, Android 8.1.0 -  missing libc.so, libart.so, boot-core-libart.oat
- Galaxy Tab A 10.1, Android 8.1.0 - missing libc.so, libflutter.so
- Samsung Galaxy J3 2017, Android 8.0.0 - missing libc.so, libflutter.so
- Sharp Aquos L2, Android 7.0 - missing libc.so, libflutter.so 
- moto g(6) plus, Android 9 - missing libc.so, libflutter.so
- Samsung Galaxy Note9, Android 9, missing libart.so, boot-core-libart.oat

Could somebody explain what is happening and advise how to build flutter application for maximum devices?

Thanks

**Comment by kf6gpe** [comment]
Does this occur with the latest stable (v.1.5.4-hotfix.2)? Did you see the same in the other channels?

**Comment by benadrasni** [comment]
It occurs with every version of flutter.

**Comment by nimesh1997** [comment]
@benadrasni So how to generate apks for both platforms( arm64 and arm) without missing libflutter.so files?

**Comment by benadrasni** [comment]
@nimesh1997 see point 4

**Comment by nimesh1997** [comment]
@benadrasni Ok so its still a issue in flutter.

**Comment by kf6gpe** [comment]
This seems to be related to the work we're doing in https://github.com/flutter/flutter/issues/31922 , which is now fixed in the dev channel and hopefully rolling to beta soon. Can you let us know if it's still a problem, please?

**Comment by benadrasni** [comment]
"flutter build appbundle" in flutter 1.8.0 now works for armeabi-v7a and 'arm64-v8a. For other platforms libflutter.so is still missing.

However I am still getting crashes with missing *.so libraries like libGLES_mali.so and libc.so, but with less frequency.

**Comment by jl-gogovapps** [comment]
I seem to be having this as well, for a Pixel 2 and Nexus 5X so far.  I've also downloaded the APKs directly from the Play Store > App Bundle Explorer > Download Device Specific APKs.  Both APKs have libflutter.so in them

<img width="609" alt="Screen Shot 2020-05-06 at 12 05 00 PM" src="https://user-images.githubusercontent.com/44532602/81200389-e4cea800-8f91-11ea-8834-027ac39c47f9.png">



```
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.4 19E287, locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
```

**Comment by xster** [comment]
Seems related to https://github.com/flutter/flutter/issues/43853. If you could show us a minimal reproduction project, it would be the most ideal. Otherwise, could you show us your build.gradle file content and the contents of a flutter build appbundle?

**Comment by TahaTesser** [comment]
Hi @benadrasni  @jl-gogovapps 
I would like to reproduce the issue, can you please minimal reproduction project as request above?  along with `flutter doctor-v` 
Thank you

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
